lib/RAST_SDK/AnnotationUtils.pm

```diff
@@ -1149,7 +1195,11 @@ sub _run_rast_workflow {
 ## process the rast genecall & annotation result
 #
 sub _post_rast_ann_call {
-    my ($self, $genome, $inputgenome, $parameters, $contigobj) = @_;
+    my ($self, $genome, $inputgenome, $rast_ref) = @_;
+
+    my %rast_details = %{ $rast_ref };
+    my $parameters = $rast_details{parameters},
+    my $contigobj = $rast_details{contigobj};
```

Comment:
You can just access those as `$rast_ref->{ parameters }` and `$rast_ref->{ contigobj }`, and any of the data within those refs as `$rast_ref->{ parameters }{ output_genome_name }` or whatever.

Reply:
Aha...easy for you to say.  I dread the $|%|@ in Perl. To avoid introducing new errors, I only changed the portion you pointed out here.

Reply:
If you are going to be continuing to work on perl code, it's well worth spending some time learning how references work and how to access data within nested data structures. In the case of nested data structures like this one, it is much easier just to deal with references and not bother dereferencing them at all; e.g.

```
$genome->{ source_id } = $rast_ref->{ parameters }{ output_genome_name };
```
I have worked on a codebase where people didn't know how to work with nested data structures (anything beyond a hash or an array), and it was a painful experience -- a bit like taking a road trip with someone who doesn't like left turns, so they only turn right or go straight ahead. The code got there in the end, but it was by a long and torturous route! Get comfortable with data structures and you will find your code becomes simpler and more concise, and easier for others who work on the codebase to understand.